static/app/stores/groupingStore.tsx

```diff
@@ -239,12 +275,53 @@ const storeConfig: Reflux.StoreDefinition & Internals & GroupingStoreInterface =
     );
 
     const responseProcessors: ResponseProcessors = {
-      merged: item => {
-        // Check for locked items
-        this.setStateForId(this.unmergeState, item.id, {
-          busy: item.state === 'locked',
+      merged: items => {
+        const newItemsMap: {[key: string]: Fingerprint} = {};
+        const newItems: Fingerprint[] = [];
+
+        items.forEach(item => {
+          if (!newItemsMap[item.id]) {
+            const newItem = {
+              id: item.id,
+              label: item.label,
+              parentId: item.parentId,
+              parentLabel: item.parentLabel,
+              eventCount: 0,
+
+              // Those properties are correct since the server returns items in
+              // descending order of lastSeen
+              lastSeen: item.lastSeen,
+              latestEvent: item.latestEvent,
+              children: [],
+            };
+            // Check for locked items
+            this.setStateForId(this.unmergeState, item.id, {
+              busy: item.state === 'locked',
+            });
+
+            newItemsMap[item.id] = newItem;
+            newItems.push(newItem);
+          }
+
+          const newItem = newItemsMap[item.id];
+          const {childId, childLabel, eventCount, lastSeen, latestEvent} = item;
+
+          if (eventCount) {
+            newItem.eventCount += eventCount;
+          }
```

Comment:
```suggestion
   items.forEach(
          ({
            id,
            label,
            parentId,
            parentLabel,
            childId,
            childLabel,
            eventCount,
            lastSeen,
            latestEvent,
            state,
          }) => {
            if (!newItemsMap[id]) {
              const newItem = {
                id,
                label,
                parentId,
                parentLabel,
                eventCount: 0,

                // Those properties are correct since the server returns items in
                // descending order of lastSeen
                lastSeen,
                latestEvent,
                children: [],
              };
              // Check for locked items
              this.setStateForId(this.unmergeState, id, {
                busy: state === 'locked',
              });

              newItemsMap[id] = newItem;
              newItems.push(newItem);
            }

            const newItem = newItemsMap[id];

            if (eventCount) {
              newItem.eventCount += eventCount;
            }
```

Reply:
this adds 10 extra lines of code, I added a counterproposal at https://github.com/getsentry/sentry/pull/25394/files#r621124130